lib/resourcebuilder/resourcebuilder.go

```diff
@@ -169,6 +169,18 @@ func (b *builder) Do(ctx context.Context) error {
 				return b.checkDeploymentHealth(ctx, actual)
 			}
 		}
+	case *batchv1.CronJob:
+		if b.modifier != nil {
+			b.modifier(typedObject)
+		}
+		if deleteReq, err := resourcedelete.DeleteCronJobv1(ctx, b.batchClientv1, typedObject,
+			updatingMode); err != nil {
+			return err
+		} else if !deleteReq {
+			if _, _, err := resourceapply.ApplyCronJobv1(ctx, b.batchClientv1, typedObject, reconcilingMode); err != nil {
```

Comment:
One wrinkle with adding official CronJob support is deciding if we want a `return b.checkCronJobHealth(ctx, actual)` like we see below for Jobs.  If we do want that, we can have a second discussion about whether we want it in this pull request, or if we'd rather file a ticket to track it as future work.

Reply:
Feels like we *should* have it but not necessarily in this PR, just making a note/card should be fine, IMO

Reply:
Good point. I will create a card regarding this issue so we can address it in the future. This pull request already improves the CVO's CronJob support and fixes the addressed issue.

Personally, at first glance, I am not sure how would the `checkCronJobHealth` method look like. And thus it would require additional effort. But as Petr said it does feel like we should have it.

Reply:
I have created [OTA-957: CVO should check the health of CronJob resources](https://issues.redhat.com/browse/OTA-957).